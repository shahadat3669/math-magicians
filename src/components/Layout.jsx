import React from 'react';
import { Outlet } from 'react-router-dom';

import Navbar from './Navbar';

const Layout = () => (
  <>
    <Navbar />
    <div className="container">
      <Outlet />
    </div>
  </>
);

export default Layout;
