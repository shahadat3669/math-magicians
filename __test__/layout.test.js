/**
 * @jest-environment jsdom
 * @jest-environment-options {"url": "https://jestjs.io/"}
 */

import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import Layout from '../src/components/Layout';

describe('Components render correctly', () => {
  it('Layout component', () => {
    const tree = renderer
      .create(
        <BrowserRouter>
          <Layout />
        </BrowserRouter>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
