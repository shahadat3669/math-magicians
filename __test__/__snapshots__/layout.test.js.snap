// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components render correctly Layout component 1`] = `
Array [
  <header
    className="header"
  >
    <nav
      className="header__navbar"
    >
      <a
        className="header__title-link"
        href="/"
        onClick={[Function]}
      >
        <h1
          className="header__title"
        >
          Math Magicians
        </h1>
      </a>
      <ul
        className="header__nav-items"
      >
        <li
          className="header__nav-item"
        >
          <a
            aria-current="page"
            className="header__nav-link active"
            href="/"
            onClick={[Function]}
          >
            Home
          </a>
        </li>
        <li
          className="header__nav-item"
        >
          <a
            className="header__nav-link "
            href="/calculator"
            onClick={[Function]}
          >
            Calculator
          </a>
        </li>
        <li
          className="header__nav-item"
        >
          <a
            className="header__nav-link"
            href="/quote"
            onClick={[Function]}
          >
            Quotes
          </a>
        </li>
      </ul>
    </nav>
  </header>,
  <div
    className="container"
  />,
]
`;
